src/sst/core/config.h

```diff
@@ -43,210 +53,283 @@ class Config : public SST::Core::Serialization::serializable
-    std::string        dump_component_graph_file; /*!< File to dump component graph */
-    bool               output_partition;   /*!< If set to true, output paritition information in config outputs */
-    std::string        output_core_prefix; /*!< Set the SST::Output prefix for the core */
-
-    RankInfo world_size;          /*!< Number of ranks, threads which should be invoked per rank */
-    uint32_t verbose;             /*!< Verbosity */
-    bool     parallel_load;       /*!< Load simulation graph in parallel */
-    bool     parallel_output;     /*!< Output simulation graph in parallel */
-    bool     no_env_config;       /*!< Bypass compile-time environmental configuration */
-    bool     enable_sig_handling; /*!< Enable signal handling */
-    bool     print_timing;        /*!< Print SST timing information */
-    bool     print_env;           /*!< Print SST environment */
+    bool setOptionFromModel(const std::string& entryName, const std::string& value);
 
-#ifdef USE_MEMPOOL
-    std::string event_dump_file; /*!< File to dump undeleted events to */
-#endif
 
-    typedef bool (Config::*flagFunction)(void);
-    typedef bool (Config::*argFunction)(const std::string& arg);
+public:
+    /** Create a new Config object.
+     * @param world_size - number of parallel ranks in the simulation
+     */
+    ~Config() {}
 
-    bool usage();
-    bool printVersion();
-    bool incrVerbose()
-    {
-        verbose++;
-        return true;
-    }
-    bool setVerbosity(const std::string& arg);
-    bool disableSigHandlers()
-    {
-        enable_sig_handling = false;
-        return true;
-    }
-    bool disableEnvConfig()
-    {
-        no_env_config = true;
-        return true;
-    }
-    bool enablePrintTiming()
-    {
-        print_timing = true;
-        return true;
-    }
-    bool enablePrintEnv()
-    {
-        print_env = true;
-        return true;
-    }
+    // Functions to access config options. Declared in order they show
+    // up in the options array
 
-#ifdef SST_CONFIG_HAVE_MPI
-    bool enableParallelLoad()
-    {
-        parallel_load = true;
-        return true;
-    }
+    // Information options
+    // No variable associated with help
+    // No variable associated with version
+
+    // Basic options
+
+    /**
+       Level of verbosity to use in the core prints using
+       Output.verbose or Output.debug.
+    */
+    uint32_t verbose() const { return verbose_; }
+
+
+    /**
+       Number of threads requested
+    */
+    uint32_t num_threads() const { return world_size_.thread; }
+
+    /**
+       Number of ranks in the simulation
+     */
+    uint32_t num_ranks() const { return world_size_.rank; }
+
+    /**
+       Name of the SDL file to use to genearte the simulation
+    */
+    const std::string& configFile() const { return configFile_; }
+
+    /**
+       Model options to pass to the SDL file
+    */
+    const std::string& model_options() const { return model_options_; }
```

Comment:
Why are we mixing snake case and word1Word2 case?

Reply:
Because I didn't feel like changing them from what they were.  It was easier to just add () to what was already there, than change the cases.